### ISSUE TITLE
Does the original developer stills alive?

### ISSUE THREAD

**Issue by leo-smi** [vatterspun/cubicexplorer#16]
I don't find anything about him on the internet :(

**Comment by leo-smi** [comment]
found the latest version [here ](http://enzo.panificiocoassin.it/esplora/cubicexplorer.html)(italian language)
[CubicExplorer 0.96.0.1582.zip](https://github.com/vatterspun/cubicexplorer/files/8157909/CubicExplorer.0.96.0.1582.zip)

**Comment by ghost** [comment]
I have always wondered what happened to the dev of the original.  Their site and everything about this software just sort of disappeared from the Internet at some point.  I've tried a ton of different file managers and never been able to duplicate the clean simplicity yet full operational capabilities of this one.  So many simple tasks this one just does ordinarily are a royal pain in everything else.  As of a month ago something happened (I don't know if it was a Windows update or what) and my CubicExplorer has started to get more and more unstable (for example, I had to turn off "infotips" -- aka tooltips over files -- because some files such as OGG Vorbis ones would completely freeze it, forcing me to kill it in the task manager because it wouldn't even do that "program has stopped responding" dialog when I clicked the X.)  I truly wish someone would pick this back up, but I guess it's dead in the water now.

I don't know if that SVN one you uploaded there is Italian, but just in case, this is a download of that same SVN version (0.96.0.1582) I got ages ago:  
[CubicExplorer_SVN.zip](https://github.com/vatterspun/cubicexplorer/files/9447972/CubicExplorer_SVN.zip)

I think that's the official ZIP from its servers at the time if that makes any sort of difference (probably not, but just in case.)  The checksums are different, so I assume there must be some difference somewhere in there.
Also, I believe this is probably the last non-SVN version in case anyone needs that:
[CubicExplorer_0.95.1.zip](https://github.com/vatterspun/cubicexplorer/files/9447981/CubicExplorer_0.95.1.zip)


I really wish someone would take up the project, but I guess we may all need to find something else soon.

**Comment by leo-smi** [comment]
I had to drop cubicexplorer, too many crashes. The best solution for now is using [QTTabBar 2048](http://qttabbar.wikidot.com/) doing some adjustments, but I miss the tree navigation. I have to say goodbye to CubicExplorer (R.I.P.).

**Comment by DerellLicht** [comment]
Yeah, once I discovered the sources here, I actually considered trying to build it myself; I have moderate experience in writing and maintaining large applications... sadly, there are two obstacles; 
1.  I'm expert in C, reasonably skilled in C++, but I don't know anything about Delphi/Pascal...
2. This project requires installing and merging a huge number of after-market libraries, which will probably be quite a challenge to get working...

It's really sad that the author just dropped the website one day, with no warning, and vanished from the earth...

**Comment by marjuanm** [comment]
Hello greetings from Mexico:

  I also used CubicExplorer when I had Windows XP on my PC and it was great, however when I changed to Windows 7 it gave a lot of problems and I stopped using it because it froze, closed by itself, etc.

I recently downloaded it for Windows 10 and it seems to work fine, it would be nice if this project was revived, I am a Visual Basic.Net programmer, I honestly don't know anything about Delphi, I don't know if it would be a good idea to port it to .NET or some other platform make it simpler, if it were in .NET I could offer to help.

**Comment by DerellLicht** [comment]
Yeah, the problem is, the Pascal code *alone* is 84,000 lines, with thousands of lines more in Components!!  Re-writing in a different language would be a *massive* task!!  I definitely don't think I'm up to it...

**Comment by ghost** [comment]
I don't think CubicExplorer is more -- or less -- buggy in Windows 7 versus 10.  In fact I have more bugs using it in 10 than I did in 7 and 8.1.

I think what we need is to somehow snag the attention of more devs.  But one thing that might possibly help on the idea there of porting it to something else might be to run it through "AI" if that's possible.  To be clear, LLM coding isn't final code and should not be actually used any more than ever is strictly necessary, but perhaps it might provide a starting point for someone with the means to try to fix it to be usable?

**Comment by DerellLicht** [comment]
Well, I think if someone really wants to resurrect this project, they will *need* to initially start with Delphi, mainly because there are so many external libraries which are required by the package; trying to switch compilers up front, would require finding replacements for all of these libraries, which would make the project FAR more complicated - and a 100,000+ line application is already complicated enough!!

Here are the external libraries which are listed in Compiling_Instructions.txt in this repository:

## Packages needed

* CEAction - http://cubicexplorer.googlecode.com/files/CEAction.zip
* PNG Components - http://cubicexplorer.googlecode.com/files/PngComponents.zip
* Tnt Controls - http://cubicexplorer.googlecode.com/files/TNTControls.zip
* Virtual Treeview - http://www.soft-gems.net
* Mustangpeak Common Library - http://www.mustangpeak.net
* EasyListview - http://www.mustangpeak.net
* VirtualShellTools - http://www.mustangpeak.net
* Toolbar2000 - http://www.jrsoftware.org/tb2k.php
* SpTBXLib - http://www.silverpointdevelopment.com/sptbxlib/
* Unicode SynEdit - http://mh-nexus.de/en/unisynedit.php
* GraphicEx - http://www.soft-gems.net
* Graphic32 - http://graphics32.org
* JCL - http://sourceforge.net/projects/jcl/
* JVCL - http://homepages.codegear.com/jedi/jvcl/

That is a *HUGE* list!!
Also, since 6 years have passed since the last CE distribution, some of these may not even be supported any more...

I've been looking at freeware Delphi variants, and there *are* some available; I've seen Lazarus used for other projects, so it has some history behind it:
https://www.lazarus-ide.org/
Also, it is still being actively supported, with a new release on 12/21/23 ...

There are others available as well, but that is one which I have seen used, which is a good start...

**Comment by firefox97** [comment]
I love Cubic Explorer.  I made a quick site to try and collect the info I could.
https://sites.google.com/view/cubicexplorer
I also though about trying to learn delphi to at least see if I could compile it.  Sometimes a really small tweak can be pretty easy if you are able to build it all...

**Comment by DerellLicht** [comment]
Well, if you could get it to build again, I would certainly be eternally grateful, and I doubt that I'm the only one... 
If you *do* attempt to build it, before trying to get an old copy of Delphi that still runs on modern OSs, I'd take a look at Lazarus, which I mentioned above... it installed very easily, and is actively supported... but I'll admit I haven't actually tried to pull CE into it yet...
